src/types/mod.rs

```diff
@@ -1,160 +1,219 @@
-        if let Some(ref s) = self.name {
-            s.clone()
-        } else {
-            format!("<op {}/{}>", self.id, self.arity)
-        }
+    pub fn variables(&self) -> Vec<Variable> {
+        (0..self.variables.len())
+            .map(|id| Variable { id })
+            .collect()
     }
-}
-
-/// Represents a symbol signifying an unspecified [`Term`].
-///
-/// [`Term`]: enum.Term.html
-#[derive(Debug, Clone, Eq)]
-pub struct Var {
-    id: DeBruijn,
-    name: Option<String>,
-}
-impl Var {
-    pub fn new(id: DeBruijn, name: Option<String>) -> Self {
-        Var { id, name }
+    pub fn new_op(&mut self, arity: u32, name: Option<String>) -> Operator {
+        let id = self.operators.len();
+        self.operators.push((arity, name));
+        Operator { id }
     }
-    pub fn name(&self) -> Option<&str> {
-        self.name.as_ref().map(|s| s.as_str())
+    pub fn new_var(&mut self, name: Option<String>) -> Variable {
+        let id = self.variables.len();
+        self.variables.push(name);
+        Variable { id }
     }
-}
-impl Hash for Var {
-    fn hash<H: Hasher>(&self, state: &mut H) {
-        self.id.hash(state);
+    /// Merge two signatures. All terms, contexts, rules, and TRSs associated
+    /// with the `other` signatures should be `reified` using methods provided
+    /// by the returned [`SignatureChange`].
+    ///
+    /// If `same_ops` is true then operators won't be added to the signature:
+    /// this must mean the operators were added in the same order for both
+    /// signatures.
+    pub fn consolidate(&mut self, mut other: Signature, same_ops: bool) -> SignatureChange {
+        let delta_op = if same_ops {
+            0
+        } else {
+            self.operators.append(&mut other.operators);
+            self.operators.len() - 1
+        };
+        let delta_var = self.variables.len();
+        self.variables.append(&mut other.variables);
+        SignatureChange {
+            delta_op,
+            delta_var,
+        }
     }
 }
-impl PartialEq for Var {
-    fn eq(&self, other: &Self) -> bool {
-        self.id == other.id
+impl Default for Signature {
+    fn default() -> Signature {
+        Signature {
+            operators: Vec::new(),
+            variables: Vec::new(),
+        }
     }
 }
-impl Ord for Var {
-    fn cmp(&self, other: &Var) -> Ordering {
-        self.id.cmp(&other.id)
+impl PartialEq for Signature {
+    fn eq(&self, other: &Signature) -> bool {
+        self.variables.len() == other.variables.len()
+            && self.operators.len() == other.operators.len()
+            && self.operators
+                .iter()
+                .zip(&other.operators)
+                .all(|(&(arity1, _), &(arity2, _))| arity1 == arity2)
     }
 }
-impl PartialOrd for Var {
-    fn partial_cmp(&self, other: &Var) -> Option<Ordering> {
-        Some(self.cmp(other))
-    }
+
+pub struct SignatureChange {
+    delta_op: usize,
+    delta_var: usize,
```

Comment:
My first reaction was to balk at the addition of a structure that clearly isn't part of the _theory_ for TRSs, but I see how this fits into the behavior of `consolidate`. As we iterate on the behavior of that function, we'll probably need to iterate on this `struct`, too. Even so, I like this addition to the library